src/gui/entry/EntryView.cpp

```diff
@@ -144,24 +268,262 @@ Entry* EntryView::entryFromIndex(const QModelIndex& index)
+ * Set state of 'Hide Usernames' setting (NOTE: just pass-through for m_model)
+ */
+void EntryView::setHideUsernames(const bool hide)
+{
+    m_model->setHideUsernames(hide);
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Get current state of 'Hide Passwords' setting (NOTE: just pass-through for
+ * m_model)
+ */
+bool EntryView::hidePasswords() const
+{
+    return m_model->hidePasswords();
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Set state of 'Hide Passwords' setting (NOTE: just pass-through for m_model)
+ */
+void EntryView::setHidePasswords(const bool hide)
+{
+    m_model->setHidePasswords(hide);
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Get current state of view
+ */
+QByteArray EntryView::viewState() const
+{
+    return header()->saveState();
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Set state of entry view view
+ */
+bool EntryView::setViewState(const QByteArray& state) const
+{
+    return header()->restoreState(state);
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Sync checkable menu actions to current state and display header context
+ * menu at specified position
+ */
+void EntryView::showHeaderMenu(const QPoint& position)
+{
+    /* Sync checked state of menu actions to current state of view */
+    m_hideUsernamesAction->setChecked(m_model->hideUsernames());
+    m_hidePasswordsAction->setChecked(m_model->hidePasswords());
+    foreach (QAction *action, m_columnActions->actions()) {
+        if (static_cast<QMetaType::Type>(action->data().type()) != QMetaType::Int) {
+            Q_ASSERT(false);
+            continue;
+        }
+        int colidx = action->data().toInt();
+        bool hidden = header()->isSectionHidden(colidx) || (header()->sectionSize(colidx) == 0);
+        action->setChecked(!hidden);
+    }
+
+    /* Display menu */
+    m_headerMenu->popup(mapToGlobal(position));
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Toggle visibility of column referenced by triggering action
+ */
+void EntryView::toggleColumnVisibility(QAction *action)
+{
+    /*
+     * Verify action carries a column index as data. Since QVariant.toInt()
+     * below will accept anything that's interpretable as int, perform a type
+     * check here to make sure data actually IS int
+     */
+    if (static_cast<QMetaType::Type>(action->data().type()) != QMetaType::Int) {
+        Q_ASSERT(false);
+        return;
+    }
+
+    /*
+     * Toggle column visibility. Visible columns will only be hidden if at
+     * least one visible column remains, as the table header will disappear
+     * entirely when all columns are hidden, rendering the context menu in-
+     * accessible
+     */
+    int colidx = action->data().toInt();
+    if (action->isChecked()) {
+        header()->showSection(colidx);
+        if (header()->sectionSize(colidx) == 0) {
+            header()->resizeSection(colidx, header()->defaultSectionSize());
+        }
+    }
+    else {
```

Comment:
return at the end of the if block, remove else.

Reply:
I wouldn't recommend that here, as this would unnecessarily disguise the inherent symmetry.

Reply:
I wouldn't worry too much about symmetry, but more about clarity and prevention of mistakes. A return here makes it clear that we don't do anything further, end of journey. You don't need to worry about what comes next anymore. It keeps the function code short and concise and the indentation level low. That every boolean value has a negative is a trivial fact.

Reply:
Yeah, clarity... We're basically on the same page, but seem to have rather different views on what clarity and readability. Well, I don't have to maintain it, so I'll just change it to your liking.

Reply:
Done

Reply:
Thanks!